pkg/controller/statusmanager/ipsec_status.go

```diff
@@ -0,0 +1,151 @@
+package statusmanager
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/openshift/cluster-network-operator/pkg/network"
+	"github.com/openshift/cluster-network-operator/pkg/platform"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func (status *StatusManager) SetFromMachineConfigs() {
+	status.Lock()
+	defer status.Unlock()
+
+	master, err := status.ipsecMachineConfigExists(platform.MasterRoleMachineConfigLabel)
+	if err != nil {
+		log.Printf("failed to retrieve machine configs for master role: %v", err)
+	}
+	worker, err := status.ipsecMachineConfigExists(platform.WorkerRoleMachineConfigLabel)
+	if err != nil {
+		log.Printf("failed to retrieve machine configs for worker role: %v", err)
+	}
```

Comment:
This should be more generic. The status manager should become aware of the specific MachineConfigs that we need rendered regardless of them being for ipsec or any other potential thing in the future, just as, for example, it is aware of the needed daemon sets. deployments, etc. So other controllers need to add that information to the status manager or we can rely on the owner reference. 

It should also check if those machine configs are been rendered into the pool or no, not just if the pool is degraded or progressing.